### ISSUE TITLE
[Youtube][Intermittent] A different resolution is set for videos either too high or too low

### ISSUE THREAD

**Issue by Softvision-RemusDranca** [MozillaReality/FirefoxReality#2946]
**[Affected Versions]:**
- Firefox Reality version 10-40720037 (dbb8ed3a)
- Firefox Reality version 9 rc10 (324719f).

**[Affected Platforms]:**
- HTC Vive Focus Plus - Android 7.1.2
- Oculus Quest - Android 7.1.1
- Pico Neo2 - Android 8.1.0

**[Prerequisites]:**
- The Firefox Reality browser is opened.

**[Steps to reproduce]:**
1. Navigate to https://www.airpano.com/360video_list.php.
2. Play the first video. (it redirects to Youtube)
3. Observe the behavior.

**[Expected results]:**
- The video is played correctly at the correct resolution.

**[Actual results]:**
- The video is set to be played at a 5k resolution and fails to play.

**[Notes]:**
- In some attempts the video Quality was set at a 360p Resolution.
- Attached are screen recordings of the issue: 
	- To high Quality: [link](https://drive.google.com/open?id=1dwxApXnoKMGalUcNcPEYWDsOEAf228V5).
	- To low Quality: [link](https://drive.google.com/open?id=1BFukJ78Xnc6KQK3gdLY3xnyeQx-5vzLV).

**Comment by daron-walters** [comment]
Still researching but [this link](https://developers.google.com/youtube/iframe_api_reference#cueVideoById) and search 'Revision History' it says that some of the methods we use have been deprecated.  still researching to see if that affects our playback but the intermittent nature of this makes me think there may be some issues on the youtube side of it.

**Comment by daron-walters** [comment]
It seems the intermittent nature of this issue is because of some issues within youtube interval calls. I have included 2 different debug sessions that were called from the same url.  Notice the call getPlaybackQuality is giving 2 different qualities from the same video.  Also it must be of note that the video of "Christ The Redemer" says 12k but it seems youtube does not support 12k yet (both in our app and on Desktop)


![dctm-2946-dev-shot-2](https://user-images.githubusercontent.com/1936826/81836560-9e3cf880-9511-11ea-8203-8534d1857097.PNG)
![dctm-2946-dev-shot-1](https://user-images.githubusercontent.com/1936826/81836570-a2691600-9511-11ea-947b-47b31d169ea0.PNG)

**Comment by keianhzo** [comment]
@Softvision-RemusDranca THis should have been fixed by https://github.com/MozillaReality/FirefoxReality/pull/3288

**Comment by Softvision-RemusDranca** [comment]
@keianhzo I've retested this issue using Firefox Reality version 11-141402320 (eda7d95) and here are the results:
	- The issue where the video is set to be played at a 5k resolution and fails to play is not reproducible on the random videos tested.
	- However, I encountered the issue where some videos where Auto played at a video Quality with a very low resolution, even if higher resolutions are available. Not sure if this is controlled by Youtube or it could have something to do with a slow Internet connection.

**Comment by daron-walters** [comment]
youtube auto corrects based on certain things and slow connection is one of those things.  There are some listeners we can take advantage of to try to help this issue

**Comment by keianhzo** [comment]
@Softvision-RemusDranca https://github.com/MozillaReality/FirefoxReality/pull/3394 forces a quality override after a video playback. Before that we were only overriding the first time we hit a watching page so it should have fixed this issue. Can you confirm and if it's not the case attach STRs so we can reproduce?

**Comment by Softvision-RemusDranca** [comment]
@keianhzo I've retested this issue using Firefox Reality version 11-141550214 (97a9dca) and I haven't encountered it while testing with multiple videos.
However, I've observed that on Oculus Quest and Pico Neo2 I can no longer change the resolution manually, for example if I want a lower one. Is this intended? 
On the other hand this behavior is not reproducible on HTC Vive Focus, the resolution can be changed manually.
